### DISCUSSION TITLE
push_gateway

### DISCUSSION THREAD

**Discussion by dradenvandewind** [prometheus/client_python#786] [Q&A]
hello,

I want push my custom-metric in promotheus in using prometheus-client  but it is not work .

Today, I have this error
  | Get "http://localhost:8000/metrics": dial tcp 127.0.0.1:8000: connect: connection refused
-- | --
how debug it ?

I find a simple example " hello world" ( docker-compose file ) to push a custom metric

regards

**Comment by csmarchbanks** [comment]
Would you provide some code for what you are trying to achieve? It looks like you may be trying to scrape a server that is not running in that error, but your title talks about push gateway.

**Reply by dradenvandewind** [reply]
here my test code


cat test/prometheus-custom-collector/code/mycollector.py
import time
import http.server
from prometheus_client.core import GaugeMetricFamily, REGISTRY, CounterMetricFamily
from prometheus_client import start_http_server
from prometheus_client import start_wsgi_server
from prometheus_client import push_to_gateway

class ServerHandler(http.server.BaseHTTPRequestHandler):
  def do_GET(self):
    self.send_response(200)
    self.end_headers()
    self.wfile.write(b"Hello World!")




class CustomCollector(object):
    def __init__(self):
        pass

    def collect(self):
        g = GaugeMetricFamily("MemoryUsage", 'Help text', labels=['instance'])
        g.add_metric(["instance01.us.west.local"], 20)
        yield g

        c = CounterMetricFamily("HttpRequests", 'Help text', labels=['app'])
        c.add_metric(["example"], 2000)
        yield c

        d = CounterMetricFamily("loudness", 'Help text', labels=['audio_monitoring'])
        d.add_metric(["loudness"],-5.62806341780557)
        yield d



if __name__ == '__main__':
    start_http_server(8000)
    #start_wsgi_server(8000)
    REGISTRY.register(CustomCollector())
    #push_to_gateway('localhost:8000', job='batchA', registry=REGISTRY)
    #REGISTRY.register(CustomCollector())
    server = http.server.HTTPServer(('', 8001), ServerHandler)
    print("Prometheus metrics available on port 8000 /metrics")
    print("HTTP server available on port 8001")
    server.serve_forever()




    while True:
        time.sleep(1)

**Reply by csmarchbanks** [reply]
Thanks! That code looks like it will work in order for Prometheus to be able to scrape the server. What does your Prometheus configuration look like as well? If you are running both this service and prometheus in docker-compose it might be necessary to scrape something like "my-service:8000" instead of "localhost:8000".

**Reply by dradenvandewind** [reply]
thanks a lot.  I have replace "localhost:8000  by my-service:8000 after set container name at "my-service". it seems correct for this moment.

**Comment by dradenvandewind** [comment]
my metrics is not available in promotheus
